ElementX/Sources/Services/ElementCall/ElementCallWidgetDriver.swift

```diff
@@ -0,0 +1,171 @@
+                                                                                 parentUrl: nil,
+                                                                                 hideHeader: nil,
+                                                                                 preload: nil,
+                                                                                 fontScale: nil,
+                                                                                 appPrompt: false,
+                                                                                 skipLobby: true,
+                                                                                 confineToRoom: true,
+                                                                                 fonts: nil,
+                                                                                 analyticsId: nil)) else {
+            return .failure(.failedBuildingWidgetSettings)
+        }
+        
+        guard let urlString = try? await generateWebviewUrl(widgetSettings: widgetSettings, room: room,
+                                                            props: .init(clientId: clientID,
+                                                                         languageTag: nil,
+                                                                         theme: nil)) else {
+            return .failure(.failedBuildingCallURL)
+        }
+        
+        guard let url = URL(string: urlString) else {
+            return .failure(.failedParsingCallURL)
+        }
+        
+        guard let widgetDriver = try? makeWidgetDriver(settings: widgetSettings) else {
+            return .failure(.failedBuildingWidgetDriver)
+        }
+        
+        self.widgetDriver = widgetDriver
+        
+        Task.detached { [weak self, widgetDriver, messagePublisher] in
+            MXLog.debug("Started message receiving loop")
+            
+            defer {
+                MXLog.debug("Stopped message receiving loop")
+            }
+            
+            while true {
+                guard let receivedMessage = await widgetDriver.handle.recv() else {
+                    return
+                }
+                
+                messagePublisher.send(receivedMessage)
+                MXLog.debug("Received message: \(receivedMessage)")
+                
+                self?.interceptAndProcessMessage(receivedMessage)
+            }
+        }
+        
+        Task.detached { [widgetDriver] in
+            MXLog.debug("Started widget driver")
+            
+            defer {
+                MXLog.debug("Stopped widget driver")
+            }
+            
+            await widgetDriver.driver.run(room: room, permissionsProvider: self)
+        }
+        
+        return .success(url)
+    }
+    
+    func processMessage(_ message: String) async -> Result<Bool, ElementCallWidgetDriverError> {
```

Comment:
This would be clearer that its client -> server if it was (and disambiguates with `interceptAndProcess` too)
```suggestion
    func sendMessage(_ message: String) async -> Result<Bool, ElementCallWidgetDriverError> {
```